base_bank_branch/models/res_bank.py

```diff
@@ -0,0 +1,83 @@
+from odoo import models, fields, api
+from odoo.osv import expression
+
+
+class ResBank(models.Model):
+    """Extend to add branches support for banks."""
+
+    _inherit = 'res.bank'
+
+    is_branch = fields.Boolean()
+    parent_id = fields.Many2one('res.bank', "Main Bank")
+    child_ids = fields.One2many('res.bank', 'parent_id', "Branches")
+    # Technical field used to know if any bank has any childs/branches
+    # as Odoo view is clunky when you need to use o2m field in domain
+    # whether its empty or not..
+    has_childs = fields.Boolean(
+        compute='_compute_has_childs',
+        string="Has Branches"
+    )
+
+    @api.depends('child_ids')
+    def _compute_has_childs(self):
+        for bank in self:
+            bank.has_childs = bool(bank.child_ids)
+
```

Comment:
No recursion check? Same bank can be branch and parent at the same time?
Can there be more than one hierarchy level?

Reply:
No, there can't be more hierarchy levels.